### PR TITLE
Integrate DeviceProfile Resources / Commands Name Change

### PR DESCRIPTION
**Pull request by tsconn23**
#234 

Please give this PR priority for approval and merge so that I can integrate it with the device-random service in order to functionally test https://github.com/edgexfoundry/edgex-go/pull/1240

Signed-off-by: Trevor Conn <trevor_conn@dell.com>

### CODE REVIEW

File: service.go [hunk 1]
```diff
@@ -196,7 +196,7 @@ func makeNewAddressable() (*models.Addressable, error) {
 			common.LoggingClient.Info(fmt.Sprintf("Addressable %s doesn't exist, creating a new one", common.ServiceName))
 			millis := time.Now().UnixNano() / int64(time.Millisecond)
 			addr = models.Addressable{
-				BaseObject: models.BaseObject{
+				Timestamps: models.Timestamps{
```
Comment (tobiasmo1) [inline service.go]: Is use of Timestamps model intended to be included as part of this changeset?

Comment (tsconn23) [reply service.go]: Yes.

**Comment by tsconn23** [comment]
recheck

**Comment by tsconn23** [comment]
recheck

File: service.go (hunk 1 above)
Comment (tobiasmo1) [reply service.go]: Thanks, I see it preceded these changes.
LGTM;

Comment (tsconn23) [reply service.go]: FYI -- Found through testing that entries in `example/cmd/device-simple/res/Simple-Driver.yaml` also needed to be changed. Verified commands are now properly created in core-metadata when device service is started.

File: internal/common/utils.go [hunk 2]
```diff
@@ -87,9 +87,9 @@ func CompareDevices(a models.Device, b models.Device) bool {
 
 func CompareDeviceProfiles(a models.DeviceProfile, b models.DeviceProfile) bool {
 	labelsOk := CompareStrings(a.Labels, b.Labels)
-	cmdsOk := CompareCommands(a.Commands, b.Commands)
+	cmdsOk := CompareCommands(a.CoreCommands, b.CoreCommands)
 	devResourcesOk := CompareDeviceResources(a.DeviceResources, b.DeviceResources)
-	resourcesOk := CompareResources(a.Resources, b.Resources)
+	resourcesOk := CompareResources(a.DeviceCommands, b.DeviceCommands)
```
Comment (jduranf) [inline internal/common/utils.go]: Should be fine this change:
`CompareCommands` → `CompareCoreCommands`
`CompareResources` → `CompareDeviceCommands`

File: internal/cache/profiles.go [hunk 3]
```diff
@@ -77,8 +77,8 @@ func (p *profileCache) Add(profile models.DeviceProfile) error {
 	p.dpMap[profile.Name] = profile
 	p.nameMap[profile.Id] = profile.Name
 	p.drMap[profile.Name] = deviceResourceSliceToMap(profile.DeviceResources)
-	p.getOpMap[profile.Name], p.setOpMap[profile.Name] = profileResourceSliceToMaps(profile.Resources)
-	p.cmdMap[profile.Name] = commandSliceToMap(profile.Commands)
+	p.getOpMap[profile.Name], p.setOpMap[profile.Name] = profileResourceSliceToMaps(profile.DeviceCommands)
+	p.cmdMap[profile.Name] = commandSliceToMap(profile.CoreCommands)
```
Comment (jduranf) [inline internal/cache/profiles.go]: `getOpMap` → `dcMap`
`cmdMap` → `ccMap`

File: internal/provision/profiles.go [hunk 4]
```diff
@@ -105,7 +105,7 @@ func profileSliceToMap(profiles []models.DeviceProfile) map[string]models.Device
 }
 
 func CreateDescriptorsFromProfile(profile *models.DeviceProfile) {
-	prs := profile.Resources
+	prs := profile.DeviceCommands
 	for _, pr := range prs {
```
Comment (jduranf) [inline internal/provision/profiles.go]: `prs` → `dcs`
`pr` → `dc`

**Pull request merged**